protos/datamodel/costs.proto

```diff
@@ -0,0 +1,236 @@
+  float sharePercentage = 6;
+
+  // Usage metrics for cost correlation (from DCGM)
+  float usageAvg = 7;
+  float usageMax = 8;
+}
+
+// StorageCostDetail represents persistent volume cost information
+message StorageCostDetail {
+  // Volume identifier
+  string volumeId = 1;
+
+  // Storage class
+  string storageClass = 2;
+
+  // Provider resource ID (for cloud storage)
+  string providerResourceId = 3;
+
+  // Storage cost
+  CostMetrics cost = 4;
+
+  // Storage usage in byte-hours
+  float byteHours = 5;
+}
+
+// LoadBalancerCostDetail represents load balancer cost information
+message LoadBalancerCostDetail {
+  // Load balancer identifier
+  string loadBalancerId = 1;
+
+  // Associated service name
+  string serviceName = 2;
+
+  // Load balancer IP address
+  string ipAddress = 3;
+
+  // Whether this is a private load balancer
+  bool isPrivate = 4;
+
+  // Load balancer cost
+  CostMetrics cost = 5;
+
+  // Hours the load balancer was active
+  float hours = 6;
+}
+
+// AllocationCost represents complete cost breakdown for a resource allocation
+// This is the main cost container that gets embedded in resource messages
+message AllocationCost {
+  // Window for this cost calculation (inherits from cluster window if not specified)
+  TimeWindow window = 1;
+
+  // Resource-specific costs
+  ResourceCost cpu = 2;
+  ResourceCost ram = 3;
+
+  // GPU costs (can have multiple GPUs)
+  repeated GPUCostDetail gpu = 4;
+
+  // Storage costs (persistent volumes)
+  repeated StorageCostDetail storage = 5;
+
+  // Network costs with detailed breakdown
+  DetailedNetworkCost network = 6;
+
+  // Load balancer costs
+  repeated LoadBalancerCostDetail loadBalancers = 7;
+
+  // Shared costs allocated to this resource
+  CostMetrics shared = 8;
+
+  // External cloud costs allocated to this resource
+  CostMetrics external = 9;
+
+  // Pre-computed total cost for performance (sum of all above)
+  CostMetrics total = 10;
+}
+
+// CostAttribution represents hierarchical cost attribution information
+// Maps to OpenCost's AllocationProperties for cost attribution
+message CostAttribution {
+  // Cluster-level attribution
+  string cluster = 1;
+  string providerId = 2;
+
+  // Node-level attribution
+  string node = 3;
+
+  // Namespace-level attribution
+  string namespace = 4;
+  map<string, string> namespaceLabels = 5;
+  map<string, string> namespaceAnnotations = 6;
+
+  // Workload-level attribution
+  string controller = 7;
+  string controllerKind = 8; // deployment, statefulset, daemonset, job
+
+  // Pod-level attribution
+  string pod = 9;
+
+  // Container-level attribution
+  string container = 10;
+
+  // Service-level attribution
+  repeated string services = 11;
+
+  // Custom attribution
+  map<string, string> labels = 12;
+  map<string, string> annotations = 13;
+}
+
+// SharedCostAllocation represents shared/overhead costs
+// For costs that need to be distributed across multiple resources
+message SharedCostAllocation {
```

Comment:
We can delete this. Shared costs are a server-side concept, applied at query time.